rpc/transaction.go

```diff
@@ -337,25 +347,25 @@ func (r *ExecutionResources) MarshalJSON() ([]byte, error) {
 			Poseidon:    NumAsHex(r.Poseidon),
 		})
 	}
-	type resources ExecutionResources // Avoid infinite recursion with MarshalJSON.
+	type resources ComputationResources // Avoid infinite recursion with MarshalJSON.
 	return json.Marshal(resources(*r))
 }
 
 // https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L1871
 type TransactionReceipt struct {
-	Type               TransactionType     `json:"type"`
-	Hash               *felt.Felt          `json:"transaction_hash"`
-	ActualFee          *FeePayment         `json:"actual_fee"`
-	ExecutionStatus    TxnExecutionStatus  `json:"execution_status"`
-	FinalityStatus     TxnFinalityStatus   `json:"finality_status"`
-	BlockHash          *felt.Felt          `json:"block_hash,omitempty"`
-	BlockNumber        *uint64             `json:"block_number,omitempty"`
-	MessagesSent       []*MsgToL1          `json:"messages_sent"`
-	Events             []*Event            `json:"events"`
-	ContractAddress    *felt.Felt          `json:"contract_address,omitempty"`
-	RevertReason       string              `json:"revert_reason,omitempty"`
-	ExecutionResources *ExecutionResources `json:"execution_resources,omitempty"`
-	MessageHash        string              `json:"message_hash,omitempty"`
+	Type               TransactionType       `json:"type"`
+	Hash               *felt.Felt            `json:"transaction_hash"`
+	ActualFee          *FeePayment           `json:"actual_fee"`
+	ExecutionStatus    TxnExecutionStatus    `json:"execution_status"`
+	FinalityStatus     TxnFinalityStatus     `json:"finality_status"`
+	BlockHash          *felt.Felt            `json:"block_hash,omitempty"`
+	BlockNumber        *uint64               `json:"block_number,omitempty"`
+	MessagesSent       []*MsgToL1            `json:"messages_sent"`
+	Events             []*Event              `json:"events"`
+	ContractAddress    *felt.Felt            `json:"contract_address,omitempty"`
+	RevertReason       string                `json:"revert_reason,omitempty"`
+	ExecutionResources *ComputationResources `json:"execution_resources,omitempty"`
```

Comment:
Is this not [Execution resources](https://github.com/starkware-libs/starknet-specs/blob/v0.7.0-rc1/api/starknet_api_openrpc.json#L2889)?

Reply:
I think it's[ function invocation](https://github.com/starkware-libs/starknet-specs/blob/v0.7.0-rc1/api/starknet_trace_api_openrpc.json#L374) that uses computation resources for the "execution resources" tag